Chinsky touches medical things

comma opened the pull request:
Moved var from human into organ. Was mostly used there.
Made blood levels defines rathe than constants hidden in one organ file. Easier to search.

Brain damage now has check to prevent paralyselocking you. Slightly lowered prob of paralysis on no-totally-fucked level of brainloss.

Lungs do not return flat oxyloss based on their damage state anymore. Instead the needed oxygen pressure is adjusted more aggressively. In effect this means that busted lung is fairly surviveable as long as you can get high-pressure internals.
Moved suffocation grace period out of damage-dealing proc into breathing handling instead. Was probably screwing over anything that was tring to do breathing damage. Better place for it either way. Losebreath also bypasses it, so busted lung gaspng is actually meaningful.

Inapto now lets you heal down from 60 or less brain damage. It also slows down brain damage taking, becoming less effective on higher stages of oxygen depravation though.

Changes way brain damage is reported on health anayzers. Now it uses damage thresholds rather than staright up values. Damage that will self heal with no meds isn't reported, damage that can be healed with inapro is 'minor', then it's weak brain activity, then extremely weak.

Made lots of places use get_shock instead of getHalloss so they're affected by stuff like painkillers.

Moved toxloss/cloneloss pain counting into the limb-level pain procs rather than overall halloss. As a side effect you can actually tell if your innards hurt if they hurt enough.

Robotic internal organ only count half of their damage for toxloss.

Overall, damages give less pain. At the same time, when damage is just applied, an amount of pain applied at the same time, so overall it's same pain as it is currently on first tick, slowly drifting towards less pain. Prevents some KOlocks by pain levels.

Alcohol now adds painkilling chem effect instead of specific check for itself.

Adrenaline now stays longer, but only does superpainkilling for 2 ticks, and usual one for 10 ticks, then it's down to just rising pulse.

comma committed:
Oxyloss and brainloss mucking

Moved var from human into organ. Was mostly used there.
Made blood levels defines rathe than constants hidden in one organ file. Easier to search.

Brain damage now has check to prevent paralyselocking you. Slightly lowered prob of paralysis on no-totally-fucked level of brainloss.

Lungs do not return flat oxyloss based on their damage state anymore. Instead the needed oxygen pressure is adjusted more aggressively. In effect this means that busted lung is fairly surviveable as long as you can get high-pressure internals.
Moved suffocation grace period out of damage-dealing proc into breathing handling instead. Was probably screwing over anything that was tring to do breathing damage. Better place for it either way. Losebreath also bypasses it, so busted lung gaspng is actually meaningful.

Inapto now lets you heal down from 60 or less brain damage. It also slows down brain damage taking, becoming less effective on higher stages of oxygen depravation though.

Changes way brain damage is reported on health anayzers. Now it uses damage thresholds rather than staright up values. Damage that will self heal with no meds isn't reported, damage that can be healed with inapro is 'minor', then it's weak brain activity, then extremely weak.

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -311,7 +311,6 @@
 	var/obj/item/organ/internal/lungs/L = internal_organs_by_name[species_organ]
 	if(!L || nervous_system_failure())
 		failed_last_breath = 1
```
breath_fail_ratio and failed_last_breath are quite similar in purpose. Why did you decide to leave failed_last_breath outside of the lungs?

comma replied:
thought it was local var of this proc, might as well axe it too

PurpleMartinJCK replied:
It's used in the breathing procs a lot with alerts and stuff.

comma replied:
Yeah looks like it might as well stay in mob, mostly used there

PurpleMartinJCK commented on code/__defines/damage_organs.dm:
```diff
@@ -63,3 +63,9 @@
+#define BLOOD_VOLUME_SAFE    85
+#define BLOOD_VOLUME_OKAY    75
+#define BLOOD_VOLUME_BAD     60
+#define BLOOD_VOLUME_SURVIVE 0
```
Did you intend to make this 0?

comma replied:
Whoosp copypaste error

PurpleMartinJCK commented:
This has a lot of stuff I've been working with and there's a few things I noticed but I don't have time to do the review at this moment. Give me until tomorrow to do one before you merge this please.

PurpleMartinJCK commented on code/modules/organs/internal/brain.dm:
```diff
@@ -183,33 +183,39 @@
 				owner.Paralyse(3)
-
+			var/can_heal = damage && (damage % damage_threshold_value || owner.chem_effects[CE_BRAIN_REGEN] || (!past_damage_threshold(3) && owner.chem_effects[CE_STABLE]))
+			var/damprob = 100
```
Why do you initialise this as 100 when you're guaranteed to write over it on line 197 and don't use it this early?

comma replied:
no reason really

PurpleMartinJCK commented on code/modules/organs/subtypes/nabber.dm:
```diff
@@ -86,8 +86,11 @@
-
-	if((acetone_volume_raw < acetone_level || !acetone_volume_raw) && owner.breath_fail_ratio < 0.25)
+	var/breath_fail_ratio = 1
+	var/obj/item/organ/internal/lungs/nabber/lungs = owner.internal_organs_by_name[BP_TRACH]
```
Please don't call the var lungs here. I know it's naming bullshit but it's important to me -_-

PurpleMartinJCK commented on code/modules/organs/subtypes/nabber.dm:
```diff
@@ -110,10 +113,10 @@
-	if(H.breath_fail_ratio >= 0.25)
-		H.adjustOxyLoss(HUMAN_MAX_OXYLOSS * H.breath_fail_ratio)
+	if(breath_fail_ratio >= 0.25)
+		H.adjustOxyLoss(HUMAN_MAX_OXYLOSS * breath_fail_ratio)
```
This here and also the human piece here https://github.com/comma/Baystation12/blob/059333285e2d5558c4aa62d56b71e0e59e3e270f/code/modules/organs/internal/lungs.dm#L253 need to be updated to respect the two minute buffer.

comma committed:
Pain changes

Made lots of places use get_shock instead of getHalloss so they're affected by stuff like painkillers.

Moved toxloss/cloneloss pain counting into the limb-level pain procs rather than overall halloss. As a side effect you can actually tell if your innards hurt if they hurt enough.

Robotic internal organ only count half of their damage for toxloss.

Overall, damages give less pain. At the same time, when damage is just applied, an amount of pain applied at the same time, so overall it's same pain as it is currently on first tick, slowly drifting towards less pain. Prevents some KOlocks by pain levels.

Alcohol now adds painkilling chem effect instead of specific check for itself.

Adrenaline now stays longer, but only does superpainkilling for 2 ticks, and usual one for 10 ticks, then it's down to just rising pulse.

PurpleMartinJCK commented:
You added more bits to this too. I'll go through it as well.

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -673,7 +673,7 @@
 				qdel(a)
 
-		if(getHalLoss() >= (species.total_health - 100))
+		if(get_shock() >= (species.total_health - 100))
```
So you're now using an instantaneous shock value instead of halloss for things in general. Why use this instead of shock_stage for things? In general? As with using the instantaneous shock, your reaction to things may not match up with where shock_stage currently is.

comma replied:
>Made lots of places use get_shock instead of getHalloss so they're affected by stuff like painkillers.

Because shock is literally halloss but with painkillers factored. I thought about merging them entirely, but few place work better with actual pain values.

PurpleMartinJCK replied:
Thing is it's impossible to get to the top stage of shock without being permanently unconscious due to pain. May want to make it -50 instead of -100 or even *.75

comma replied:
That would affect balance of tasers and other non lethal stuff (which is main client of this check) so I'd rather not touch that if it can be avoided. Making painkillers work is already going too hot.

PurpleMartinJCK replied:
Yeah ok

PurpleMartinJCK commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Medicine.dm:
```diff
@@ -650,17 +650,18 @@
 	color = "#C8A5DC"
 	scannable = 1
 	overdose = 10
+	metabolism = 0.1
```
Other chemicals generally use REM factors in their metabolism.

comma replied:
And others don't.

PurpleMartinJCK replied:
Is there a policy on this @PsiOmegaDelta?

PurpleMartinJCK commented on code/modules/organs/internal/_internal.dm:
```diff
@@ -75,4 +75,9 @@
\ No newline at end of file
+	min_broken_damage += 10
+
+/obj/item/organ/internal/proc/getToxLoss()
```
This needs to also be used in the main one in human_damage https://github.com/comma/Baystation12/blob/671140597e70595a08b2bbb6fbc01eabbfce6d18/code/modules/mob/living/carbon/human/human_damage.dm#L170

comma replied:
updooted

PurpleMartinJCK commented on code/modules/organs/organ.dm:
```diff
@@ -411,3 +411,6 @@ var/list/organ_cache = list()
 	if(rejecting)
 		. += "Genetic Rejection"
+		
+/obj/item/organ/proc/isrobotic()
```
These are used in a lot of places. Try to convert as many as you can find. There are a lot in surgery.

comma replied:
I'd rather not, it'll bloat a diff to infinity because those are indeed used in bajillion places. Best left for a separate PR.

PurpleMartinJCK replied:
Fair enough

PurpleMartinJCK commented on code/modules/organs/external/_external_damage.dm:
```diff
@@ -212,7 +214,10 @@
-	return pain + lasting_pain + 1.2 * brute_dam + 1.5 * burn_dam
+	var/tox_dam = 0
+	for(var/obj/item/organ/internal/I in internal_organs)
+		tox_dam += I.getToxLoss()
```
You may want to add a minimum threshold before you add it into pain. Like 5 or 10.

comma replied:
It's already factored by 0.3, so should be ok (5 or 10 would be ~2 or ~3). During testing lungs only started slightly showing up on mob health dolly after bruised stage.

PurpleMartinJCK replied:
It'll definitely help people tell when their organs are all dying.

PurpleMartinJCK commented on code/modules/organs/subtypes/nabber.dm:
```diff
@@ -86,8 +86,11 @@
-
-	if((acetone_volume_raw < acetone_level || !acetone_volume_raw) && owner.breath_fail_ratio < 0.25)
+	var/breath_fail_ratio = 1
+	var/obj/item/organ/internal/lungs/nabber/totally_not_lungs_I_swear = owner.internal_organs_by_name[BP_TRACH]
```
-_-

comma replied:
hey you asked

PurpleMartinJCK replied:
Chinsky plz

comma replied:
Not changing it anymore, it's var used in this and next line only so name matters little.

PurpleMartinJCK replied:
Yeah I'm actually fine with this one.

PurpleMartinJCK commented on html/changelogs/chinsky - brainstuff.yml:
```diff
@@ -0,0 +1,6 @@
+author: Chinsky
```
Please update with pain changes

PurpleMartinJCK replied:
I added the tag for it now too

comma replied:
I already added painstuff in there, too slow. Not a lot changes that would require changelog.

PurpleMartinJCK replied:
Maaaaan

PurpleMartinJCK commented on code/modules/mob/living/carbon/human/human_damage.dm:
```diff
@@ -377,7 +376,7 @@ This function restores all organs.
 		damage *= blocked_mult(blocked)
 
-	if(damage > 15)
+	if(damage > 15 && prob(damage*4))
```
You already barely ever get any adrenaline for stuff. Especially not smaller wounds. I'd say if you're adding a probably on it you should lower the cap or remove it. Perhaps a minimum amount added too.

comma replied:
Smaller wounds aren't supposed to get it at all, it's for OH SHIT moments sort of.
I dunno I just wanted to make it less guaranteed thing, prob is still pretty high, 60 at minimum threshold level, guaranteed at 25+ damage.

PurpleMartinJCK replied:
Yeah ok. I may mess with it later 👀

Techhead0 replied:
With the prob, I could see lowering the threshold to 10 or so. But it's fine for now.
